topics/transcriptomics/tutorials/scrna_preprocessing/tutorial.md

```diff
@@ -0,0 +1,1382 @@
+
+
+# Multiple Batches
+
+[Back to previous](javascript:window.history.back())
+
+ ![Workflow Lower](../../images/scrna_workflow_lower.png "Multiple batch processing")
+
+Handling more than one batch of sequencing data is rather trivial when we take into account our main goals and requirements:
+
+  1. For each batch, convert FASTQ reads from into a count matrix.
+  2. Merge all count matrices into a single count matrix
+
+The first step requires us to merely run the same workflow on each of our batches, using the exact same inputs except for the FASTQ paired data. The second step requires a minimal level of interaction from us; namely using a merge tool and selecting our matrices. 
+
+
+### Data upload and organization
+
+The count matrix we have generated in the previous section is too sparse to perform any reasonable analysis upon, and constitutes data only of a single batch. Here we will use more populated count matrices from multiple batches, under the assumption that we now know how to generate each individual one of them using the steps provided in the previous section. This data is available at [`Zenodo`](https://zenodo.org/record/2554612).
+
+Once again, file naming is important, and so we will rename our matrix files appropriately to the plate and batch they are supposed to originate from.
+
+> ### {% icon hands_on %} Hands-on: Data upload and organization
+>
+> 1. Create a new history and rename it (*e.g.* scRNA-seq multiple-batch tutorial)
+> 1. Import the four matrices (`P1_B1.tabular`, `P1_B2.tabular`, etc.) from [`Zenodo`](https://zenodo.org/record/2554612) or from the data library (ask your instructor)
+>    - Set the datatype of the tabular files to **tabular**
+>
+>    ```
+>    https://zenodo.org/record/2554612/files/P1_B1.tabular?download=1
+>    https://zenodo.org/record/2554612/files/P1_B2.tabular?download=1
+>    https://zenodo.org/record/2554612/files/P2_B3.tabular?download=1
+>    https://zenodo.org/record/2554612/files/P2_B4.tabular?download=1
+>    ```
+>    
+>    {% include snippets/import_via_link.md %}
+>    {% include snippets/import_from_data_library.md %}
+>    
+> 1. Rename a matrix
+>    - Click on {% icon galaxy-pencil %} of the *`P1_B1.tabular`* file
+>    - Set the Name field such that it is affixed with "_P1_B1" (e.g. 'multibatch_P1_B1')
+>    - Click *Save*
+> 1. Repeat for all matrices
+>    - **Pay attention to the Plate number which changes after Batch 4**
+>
+{: .hands_on}
+
+## Merging Count Matrices
+
+Before we begin, we must consider that our matrices are not equal -- e.g. Batch1 has at least 1 cell that maps to GeneA, whereas Batch2 has no mention of GeneA.
+
+This can be resolved by performing a "Full Join" (as described [here](http://www.sql-join.com/sql-join-types/)) where GeneA is inserted into Batch2 as a gene with 0 counts:
```

Comment:
No. With so many other topics touched and explained SQL is just clearly over the top. Explain it by example and in simple (even simplified if necessary) words, but no one in the audience will want to read additional material at this point.

Reply:
The ASCII table below explains the full join nicely, but the link also gives some really nice visual examples which would explain other types of joins for the more enthusiastic users.

I will keep the link, but place it under the ASCII tables as a footnote